### DISCUSSION TITLE
Run XDE under Dawn

### DISCUSSION THREAD

**Discussion by mmoeller86** [devhawala/dwarf#10] [Q&A]
Hi dear Dwarf-Team,

how can I run the XDE Disk Image provided on your page under Dawn ?

Greets,
Martin

**Comment by mmoeller86** [comment]
Is this simply zlib-compressed (I tried with file.exe from cygwin64) or is this a special encoding ?
When simply zlb-compressed is this enough to write a program which decompresses using
the zlib library (or have you one) ? I tried with gunzip but this didn't worked !

**Comment by mmoeller86** [comment]
I wrote a simple C Program using zlib library, it decompresses but it didn't run under Dawn.
It run about 1 minute booting time, but did it work or does it just boot ?
I will retry.

**Comment by devhawala** [comment]
Hello,

i suppose that the XDE image you mean is one in the [disks-6085](https://github.com/devhawala/dwarf/tree/master/disks-6085) directory of the project.

So no, Dawn will not be able to run it,  as the disk images in this directory are intended for my 6085 emulation (*Draco*). Besides the compression (zlib-like with Java standard functionality) with the delta files for my emulator, the block structure of the disk tracks is different and very Xerox specific, with an additional label record for each data sector. Furthermore, the bootfiles on the *Draco*-disks (germ and Pilot/XDE) require the 6085 I/O architecture (IOP-board) in part emulated by *Draco* but which Dawn does not support.

Dawn requires the younger "label-less" sector structure which was used for SCSI disks on newer physical machines (6085-II, 8090) or disk images in the case of the GVWin emulator: these are simple files with 512 bytes raw data blocks for each sector, without the additional label used before.

But the disk image provided with Dawn is in fact an XDE system. You can also run this image with the GVWin emulators, as both use the same disk image structure and emulate the same Mesa instruction set and I/O architecture (software *agents*). I'm not sure if you can run an GlobalView disk from GVWin with Dawn, i never tried...

Regards
Hans

**Reply by jpasqua** [reply]
I just ran across this repository and am finding my way around. Thank you so much for creating this. I was a Pilot kernel engineer starting in 1981 and spent a lot of my time working on the file system and volume manager back then. I'm primarily interested in the XDE emulation since that's where I spent most of my time. I am focused on duchess so that I can set the screen size. Eventually I want to adapt this to run full screen on a native 1024x768 display in a small replica.

I'm using the dawn XDE disk and germ for my testing, but it is a bare-bones installation. Before I try to recreate something more like the contents of xde5.0_2xTajo+hacks.zdisk, I thought I'd ask if that already exists somewhere in a Duchess-compatible form.

Again, thank you for your work on this. It's really quite nice.

**Reply by pamoroso** [reply]
@jpasqua In case you're interested the [Medley Interlisp Project](https://interlisp.org) is reviving the Medley system.

**Reply by devhawala** [reply]
Hello,

as far as i know, there is no Dawn disk prepared with the Tajo tools found on various floppy image sets at Bitsavers.

But Duchess can mount these floppies for reading, so you can setup a Dawn+Tools disk for Duchess similar to the xde5.0_2xTajo+hacks disk for Draco. But it's not sure that all tools will work, as there were some incompatible interface changes: there is no "DisplayImpl.bcd" with Dawn, so i tried to run ClockWithDisplayImpl.bcd from one of the disks, but it did not start...

Greetings, hans

**Reply by jpasqua** [reply]
Thanks Hans, I will have to look through the bitsavers floppy images a bit more. I downloaded a couple of sets and went through them in Duchess but didn't see all of the hacks (including a couple I wrote which I'm not sure exist anywhere at this point).

**Reply by jpasqua** [reply]
BTW, I've hacked together a proof of concept for a full screen mode that hides the tool bar and status line. They can be toggled on and off again with a key press (hardwired to F12 at the moment). When running in full screen mode I place the tool bar NORTH, the status line CENTER, and the display panel SOUTH. When the toolbar/status are toggled on, I lose the equivalent amount of space from the bottom of the display panel. This is acceptable for my purposes since I will add a physical display for MP Codes and probably less for disk and floppy accesses. I will usually only make the controls visible to shut down the emulation.
